### ISSUE TITLE
Look for config in package.json

### ISSUE THREAD

**Issue by LinusU** [babel/babel#1818]
*Sorry if this has been discussed before, my search didn't turn up anything*

I would like to make a feature request regarding config reading. It would be cool if babel could look inside the `package.json` file and find config options there. That would allow me to keep all of my config for my library at the same place.

This is, part of, my `package.json` currently.

```javascript
  "devDependencies": {
    "babel": "^5.6.4",
    "mocha": "^2.2.4",
    "standard": "^4.3.2"
  },
  "scripts": {
    "test": "standard && mocha --compilers js:babel/register",
    "build": "babel src --out-dir lib",
    "prepublish": "npm run build"
  },
  "standard": {
    "parser": "babel-eslint"
  }
```

I also have a `.babelrc` just for specifying that I want to use `es7.functionBind`. At first I just had it under `scripts.build`, but then it didn't work for `npm test` since the mocha compiler option doesn't support this.

Ideally I would like to add this to my `package.json`:

```javascript
  "babel": {
    "optional": [ "es7.functionBind" ]
  }
```

**Comment by sebmck** [comment]
>  It would be cool if babel could look inside the `package.json` file and find config options there

I'm hesitant on adding an additional way to specify options. The goal of `.babelrc` was to consolidate it.

> the mocha compiler option doesn't support this

It should. If it doesn't then it's a bug.

**Comment by LinusU** [comment]
Just to clarify, what didn't work was to give babel command line options via the `mocha --compiler` option. Using a `.babelrc` works, I would just prefer to have everything at the same place.

**Comment by Mark-Simulacrum** [comment]
How were you attempting to pass options to babel through `mocha --compiler`? I don't believe `mocha` support giving options to compilers.

However, you can use the method described on [this](http://stackoverflow.com/a/28700064/4085230) StackOverflow post to compile tests while giving `babel` options.

**Comment by LinusU** [comment]
@Mark-Simulacrum It dosen't, thats the problem. I solved it by creating a `.babelrc` file which is another answer to the very same question you posted.

It would however be nice to be able to specify the babel options right in the package.json, which many other libraries support.

I'm open for leaving the bug open for a while, to see if there is any demand for it. If I'm the only one that wants it, then we shouldn't implement it.

**Comment by LinusU** [comment]
Thanks @sebmck, should we document this behaviour as well?

**Comment by sebmck** [comment]
Yup, it'll be due out in the next minor. I've restructured all the internals and just need to make sure I don't push out a bad release.

**Comment by LinusU** [comment]
Exciting news! Good luck :+1:

**Comment by cesarandreu** [comment]
This is great! Getting rid of my `.babelrc` file means less clutter in my project's root, thanks :+1:

**Comment by sebmck** [comment]
Released as of 5.7.0, thanks! :blush:

**Comment by sahilsk** [comment]
https://babeljs.io/docs/usage/babelrc/

^ adding link so that dumb like me don't have hard time finding relevant part of the documentation
